include/boost/regex/v4/regex_workaround.hpp

```diff
@@ -207,9 +203,11 @@ namespace boost{ namespace re_detail{
       const char *strSource 
    )
    {
-      if(std::strlen(strSource) + std::strlen(strDestination) + 1 > sizeInBytes)
+      std::size_t lenSource = std::strlen(strSource) + 1;
+      std::size_t lenDestination = std::strlen(strDestination);
```

Comment:
There two ways we can resolve this. Both ways are fine by me. Be sure to let me know which one you prefer.

- Rename `lenSource` to something like `lenSourceWithNull`, or
- Don't add the `+ 1` and add it in both places where we need it.

My only concern with the second approach is that it makes the arithmetic that's going on less obvious. Right now it is quite obvious that this will never access the buffer out of bounds.